plugins/woocommerce-blocks/assets/js/data/checkout/utils.ts

```diff
@@ -230,3 +233,36 @@ export const getPaymentResultFromCheckoutResponse = (
 
 	return paymentResult;
 };
+
+export const validateAdditionalFields = (
+	additionalFields: AdditionalValues
+): boolean => {
+	// Early return if no additional fields to validate
+	if ( Object.keys( additionalFields ).length === 0 ) {
+		return true;
+	}
+
+	const validationStore = select( VALIDATION_STORE_KEY );
+
+	// Check each additional field for validation errors
+	// The validation store prefixes fields with 'additional-field-'
+	for ( const fieldKey of Object.keys( additionalFields ) ) {
+		// We don't know which location the field is in, so we need to check both
+		let error = validationStore.getValidationError(
```

Comment:
@senadir is there a better way to detect or know where a field is on client side? I feel like there will be.

Reply:
@alexflorisca Nadir shared this which can be reused instead of checking for both https://github.com/woocommerce/woocommerce/blob/74c589d6efea4003652e6b1d3eb40291549c9001/plugins/woocommerce-blocks/assets/js/settings/blocks/constants.ts#L124-L155